injector/node_failure.go

```diff
@@ -57,14 +58,21 @@ func NewNodeFailureInjector(spec v1beta1.NodeFailureSpec, config NodeFailureInje
 }
 
 // Inject triggers a kernel panic through the sysrq trigger
-func (i nodeFailureInjector) Inject() error {
+func (i nodeFailureInjector) Inject(ctx context.Context) error {
 	var err error
 
 	i.config.Log.Infow("injecting a node failure by triggering a kernel panic",
 		"sysrq_path", i.sysrqPath,
 		"sysrq_trigger_path", i.sysrqTriggerPath,
 	)
 
+	select {
+	case <-ctx.Done():
```

Comment:
This is inconsistent to everywhere else, lmk if I should stick with `select` everywhere or `if`. `select` appears to be more idiomatic (and I think clearer to read as "Done" is more sensible than "Err"), but the `if` is smaller.